pkg/http-tunnel/server/registry.go

```diff
@@ -0,0 +1,245 @@
+		"client-name": cname,
+		"client-id":   origaddr,
+		"data":        reg,
+	}).Info("REGISTRY SUBSCRIBE")
+}
+
+// GetID returns the ID for this client
+func (r *registry) GetID(cname string) string {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	v, ok := r.items[cname]
+	if !ok {
+		return ""
+	}
+	return v.ClientID
+}
+
+// Subscriber returns client identifier assigned to given host.
+func (r *registry) Subscriber(hostPort string) (string, *Auth, bool) {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	h, ok := r.hosts[trimPort(hostPort)]
+	if !ok {
+		return "", nil, false
+	}
+	fmt.Printf("SUBSCRIBER REGISTRY [%s] value : %+v \n", hostPort, h)
+
+	return h.identifier, h.auth, ok
+}
+
+// Unsubscribe removes client from registry and returns it's RegistryItem.
+func (r *registry) Unsubscribe(identifier string, idname string) *RegistryItem {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	i, ok := r.items[identifier]
+	if !ok {
+		fmt.Printf("UNSUBSCRIBE REGISTRY error not found ID [%s] Idname [%s] value : %+v \n", identifier, idname, i)
+		return nil
+	}
+	fmt.Printf("UNSUBSCRIBE REGISTRY Identifier [%s] Idname [%s] value : %+v \n", identifier, idname, i)
+
+	r.logger.WithFields(log.Fields{
+		"identifier": identifier,
+		"id-name":    idname,
+		"data":       i,
+	}).Info("REGISTRY UNSUBSCRIBE")
+
+	if i.Hosts != nil {
+		for _, h := range i.Hosts {
+			delete(r.hosts, h.Host)
+		}
+	}
+
+	delete(r.items, identifier)
+
+	return i
+}
+
+func (r *registry) set(i *RegistryItem, identifier string) error {
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	j, ok := r.items[identifier]
+	if !ok {
+		r.logger.WithFields(log.Fields{
+			"client-name": identifier,
+			"client-id":   i.ClientID,
+			"data":        i,
+		}).Error("REGISTRY SET ERROR: client-name not found")
+
+		return errClientNotSubscribed
+	}
+
+	r.logger.WithFields(log.Fields{
+		"client-name": identifier,
+		"client-id":   j.ClientID,
+		"data":        j,
+	}).Info("REGISTRY SET (OLD FOUND)")
+
+	//fmt.Printf("OLD REGISTRY Identifier [%s] value : %+v \n", identifier, j)
+	i.ClientID = j.ClientID
+
+	r.logger.WithFields(log.Fields{
+		"client-name": identifier,
+		"client-id":   i.ClientID,
+		"data":        i,
+	}).Info("REGISTRY SET (NEW SET)")
+
+	if i.Hosts != nil {
+		for _, h := range i.Hosts {
+			if h.Auth != nil && h.Auth.Token == "" {
+				return fmt.Errorf("missing auth token")
+			}
+			if _, ok := r.hosts[trimPort(h.Host)]; ok {
+				return fmt.Errorf("host %q is occupied", h.Host)
+			}
+		}
+
+		for _, h := range i.Hosts {
+			r.hosts[trimPort(h.Host)] = &hostInfo{
+				identifier: identifier,
+				auth:       h.Auth,
+			}
+		}
+	}
+	//fmt.Printf("SET REGISTRY Identifier [%s] value : %+v \n", identifier, i)
+	r.items[identifier] = i
+	r.source[i.ClientID] = i
+
+	return nil
+}
+
+func (r *registry) clear(identifier string) *RegistryItem {
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	ilogger := r.logger.WithFields(log.Fields{
+		"identifier": identifier,
+	})
+
+	//i, ok := r.items[identifier]
```

Comment:
There is a lot of unnecessary comented out code that we should look into and erase